File: webapp/HIN_WEB/Models/ContactInformation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HIN_WEB.Models
{
    public class ContactInformation
    {
        public int Id { get; set; }
        [CsvImport]
        [CsvRequired]
        public string FirstName { get; set; }
        [CsvImport]
        public string LastName { get; set; }
        public string ContactName { get { return FirstName + " " + LastName; } }
        [CsvImport]
        public string ContactTitle { get; set; }
        public string OfficeNumber { get; set; }
        public string Extension { get; set; }
        public string FaxNumber { get; set; }
        public string CellNumber { get; set; }
        [CsvImport]
        public int? Gender { get; set; }
        public bool? Inactive { get; set; }
        public string Email { get; set; }
        public int? Type { get; set; }
        public int? EntityId { get; set; }
        public string Notes { get; set; }
        public string LinkedInProfile { get; set; }
        public bool? IsAdditionalInfo { get; set; }
        public bool? IsMember { get; set; }
        public bool? IsGuest { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public string UpdatedBy { get; set; }
        public int? UpdatedById { get; set; }
        public DateTime? CreatedOn { get; set; }
        public string CreatedBy { get; set; }
        public int? CreatedById { get; set; }
        public string BatchNumber { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Country { get; set; }
        public string ZipCode { get; set; }
        public int? ContactGroupId { get; set; }
        public string ContactInfoNumber { get; set; }
        public string ContactImage { get; set; }
        public int? OrginId { get; set; }
        public int? LeadId { get; set; }
        public int? VendorId { get; set; }
        public int? PartnerId { get; set; }
        public bool? IsPrimary { get; set; }
        public string Website { get; set; }
        public string MiddleName { get; set; }
        public string SecondMiddleName { get; set; }
        public string SecondLastName { get; set; }
    }
}

File: webapp/HIN_WEB/ClientApp/src/app/contactinformation/contactinformation.ts
import { Timecost } from "../model/timecost";
import { Materialcost } from "../model/materialcost";

export class Contactinformation {
  public Id: number;
  public FirstName: string;
  public LastName: string;
  public ContactName: string;
  public MiddleName: string;
  public SecondMiddleName: string;
  public SecondLastName: string;
  public ContactTitle: string;
  public OfficeNumber: string;
  public Gender: number;
  public Inactive: boolean;
  public Extension: string;
  public FaxNumber: string;
  public CellNumber: string;
  public Email: string;
  public Type: number;
  public EntityId: number;
  public LinkedInProfile: string;
  public UpdatedOn: Date;
  public UpdatedBy: string;
  public UpdatedById: number;
  public CreatedOn: Date;
  public CreatedBy: string;
  public CreatedById: number;
  public Notes: string;
  public IsAdditionalInfo: boolean;
  public IsMember: boolean;
  public IsGuest: boolean;
  public BatchNumber: string;
  public Address: string;
  public City: string;
  public State: string;
  public Country: string;
  public ZipCode: string;
  public ContactGroupId: number;
  public ContactInfoNumber: string;
  public ContactImage: string;
  public OrginId: number;
  public LeadId: number;
  public VendorId: number;
  public PartnerId: number;
  public IsPrimary: boolean;
  public Website: string;
}
export class DealContact {
  public Id: number;
  public DealId: number;
  public EntityTypeId: number;
  public ContactId: number;
  public ContactName: string;
  public ContactTitle: string;
  public CalendarInviteType: number;
  public CompletedStatusId: number;
  public ZoomId: string;
  public CellNumber: string;
  public OfficeNumber: string;
  public Email: string;
  public AssignedTo: number;
  public SalesMan: number;
  public QuickNotes: number;
  public Notes: string;
  public StartDate: Date;
  public EndDate: Date;
  public Color: string;
  public timecosts: Array<Timecost>;
  public materialcosts: Array<Materialcost>;
  public StepTypeId: number;
  public IsSaved: boolean = false;
  public nextStepToggle: boolean = false;
  public UpdatedOn: Date;
  public UpdatedBy: string;
  public UpdatedById: number;
  public CreatedOn: Date;
  public CreatedBy: string;
  public CreatedById: number;
  public Other: string;
  public nextStep: DealContactNextStep = new DealContactNextStep();
}
export class LeadContact {
  public Id: number;
  public LeadId: number;
  public ContactId: number;
  public ContactName: string;
  public UpdatedOn: Date;
  public UpdatedBy: string;
  public UpdatedById: number;
  public CreatedOn: Date;
  public CreatedBy: string;
  public CreatedById: number;
}
export class DealContactNextStep {
  public Id: number;
  public DealId: number;
  public DealContactId: number;
  public ContactId: number;
  public ContactName: string;
  public CalendarInviteType: number;
  public ZoomId: string;
  public CellNumber: string;
  public Email: string;
  public OfficeNumber: string;
  public AssignedTo: number;
  public Notes: string;
  public EndDate: Date;
  public UpdatedOn: Date;
  public UpdatedBy: string;
  public UpdatedById: number;
  public CreatedOn: Date;
  public CreatedBy: string;
  public CreatedById: number;
  public ContactTitle: string;
  public StepTypeId: number;
  public StartDate: Date;
  public StatusId: number;
  public Salesman: number;
  public IsSaved: boolean = false;
  public QuickNotes: number;
  public ColorCode: string;
  public Other: string;
}
export class Group {
  public Id: number;
  public GroupName: string;
  public Inactive: boolean;
  public CreatedOn: Date;
  public CreatedBy: string;
  public UpdatedOn: Date;
  public UpdatedBy: string;
  public CreatedById: number;
  public UpdatedById: number;
}
export class ContactGroupModel {
  public Group: Group;
  public ContactInformations: Array<Contactinformation>;
}
export class ContactGroup {
  public Id: number;
  public GroupId: number;
  public ContactId: number;
}


export class SelectedContact {
  public Id: number;
  public EntityId: number;
  public EntityType: number;
}
